oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -47,29 +49,70 @@ public static List<Rule> loadRules(final String path) throws ModuleStartExceptio
-                    if (!enabledRules.contains(fileName)) {
+                .flatMap(f -> {
+                    if (f.isDirectory()) {
+                        return Arrays.stream(Objects.requireNonNull(f.listFiles()))
+                                .filter(File::isFile)
+                                .map(file -> {
+                                    try (Reader r = new FileReader(file)) {
+                                        String fileName = file.getName();
+                                        int dotIndex = fileName.lastIndexOf('.');
+
+                                        if (dotIndex == -1 || !"yaml".equals(fileName.substring(dotIndex + 1))) {
+                                            return null;
+                                        }
+                                        String ruleName = fileName.substring(0, dotIndex);
+                                        fileName = f.getName() + '/' + fileName;
+                                        if (!formedEnabledRules.contains(fileName) && !formedEnabledRules.contains(f.getName() + "/*.yaml")) {
+                                            return null;
+                                        }
+                                        Rule rule = new Yaml().loadAs(r, Rule.class);
+                                        if (rule == null) {
+                                            return null;
+                                        }
+                                        rule.setName(ruleName);
+                                        return rule;
+                                    } catch (IOException e) {
+                                        LOG.debug("Reading file {} failed", f, e);
+                                    }
+                                    return null;
+                                });
+                    } else {
+                        try (Reader r = new FileReader(f)) {
+                            String fileName = f.getName();
+                            int dotIndex = fileName.lastIndexOf('.');
+                            if (dotIndex == -1 || !"yaml".equals(fileName.substring(dotIndex + 1))) {
+                                return null;
+                            }
+                            String ruleName = fileName.substring(0, dotIndex);
+                            if (!formedEnabledRules.contains(fileName)) {
+                                return null;
+                            }
+                            Rule rule = new Yaml().loadAs(r, Rule.class);
+                            if (rule == null) {
+                                return null;
+                            }
+                            rule.setName(ruleName);
+                            return Stream.of(rule);
+                        } catch (IOException e) {
+                            LOG.debug("Reading file {} failed", f, e);
+                        }
```

Comment:
Can you extract this to a method and share it with the lines above in the `if` statement?